src/Utils/Logging.hs

```diff
@@ -481,37 +484,50 @@ withElasticsearchBackend mgr esServer ixName inner = do
         inner $ \a -> atomically (writeTBQueue queue a)
 
   where
+    curIxName = do
+        d <- T.pack . formatTime defaultTimeLocale "%Y.%m.%d" <$> getCurrentTime
+        return $ ixName <> "-" <> d
+
     errorLogFun Error msg = T.hPutStrLn stderr msg
     errorLogFun _ _ = return ()
 
     processor queue = do
+        i <- curIxName
         (n, msg) <- atomically $ do
             h <- readTBQueue queue
-            go (1000 :: Int) (indexAction h) (1 :: Int)
+            go i (1000 :: Int) (indexAction i h) (1 :: Int)
         errorLogFun Info $ "send " <> sshow n <> " messages"
 
+        createIndex i
         void $ HTTP.httpLbs (putBulgLog msg) mgr
       where
-        go 0 !b !c = return (c, b)
-        go i !b !c = tryReadTBQueue queue >>= \case
+        go _ 0 !b !c = return (c, b)
+        go cix i !b !c = tryReadTBQueue queue >>= \case
             Nothing -> return (c, b)
-            Just x -> go i (b <> indexAction x) (pred c)
+            Just x -> do
+                go cix i (b <> indexAction cix x) (pred c)
```

Comment:
Is it not still an issue that `i` isn't being `pred`'d here? What stops the recursion?

Reply:
I think, the code is just wrong, and always was. Not just that `i` isn't decremented, but `c` is, which should be incremented.

Currently the recursion terminates only when the queue is empty, which works but defeats the purpose of the bound. However, just decrementing the bound wouldn't make sense neither, because we would start creating larger batches only after saturating the network link or es.